internal/controller/controllers/spoke_k8s_client.go

```diff
@@ -109,9 +111,60 @@ func (c *spokeK8sClient) ApproveCsr(csr *certificatesv1.CertificateSigningReques
 }
 
 func (c *spokeK8sClient) GetNode(name string) (*corev1.Node, error) {
-	node, err := c.nodesClient.Get(context.TODO(), name, metav1.GetOptions{})
+	node := &corev1.Node{}
+	if err := c.Get(context.TODO(), types.NamespacedName{Name: name}, node); err != nil {
+		return nil, err
+	}
+	return node, nil
+}
+
+func (c *spokeK8sClient) StartAgentForReclaim(ctx context.Context, agent *aiv1beta1.Agent, image string) error {
+	hostname := getAgentHostname(agent)
+	node, err := c.GetNode(hostname)
 	if err != nil {
-		node = nil
+		return errors.Wrapf(err, "failed to find node %s", hostname)
+	}
+
+	namespaceName := "assisted-installer"
+	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}}
+	if err := c.Get(ctx, types.NamespacedName{Name: ns.Name}, ns); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return errors.Wrapf(err, "failed to get namespace %s", namespaceName)
+		}
+		if err := c.Create(ctx, ns); err != nil {
```

Comment:
Are we leaving the namespace in the spoke after reclaim ?

Reply:
In clusters we install this namespace is actually present.
It's where the controller job runs.

But also, yes. We would be leaving anything we create there.
I think this could probably be handled as a part of https://issues.redhat.com/browse/MGMT-10006